homeassistant/components/sonos/speaker.py

```diff
@@ -77,7 +77,7 @@
     "renderingControl",
     "zoneGroupTopology",
 }
-SUPPORTED_VANISH_REASONS = ("sleeping", "switch to bluetooth", "upgrade")
+SUPPORTED_VANISH_REASONS = ("powered off", "sleeping", "switch to bluetooth", "upgrade")
```

Comment:
Shouldn't they be `off` rather than `unavailable` when that happens?

Reply:
If the speaker is powered off, we don't know if it will ever come back as it is not in an "idle" state. We can't poll for state and cannot turn it back on. `Unavailable` is the proper state here.

Reply:
Oh.  I thought powered-off Sonos speakers would work kind of like my DLNA speakers or my Marantz receiver — they still allow for connection to check that they are "off" (soft off).  I see now that these are *vanishing* and therefore they are unavailable in their own right — cannot be powered back on remotely.  Thanks for the kind explanation.